packages/block-library/src/paragraph/edit.js

```diff
@@ -157,61 +109,50 @@ function ParagraphBlock( {
 						}
 					/>
 				</PanelBody>
-				<ParagraphPanelColor
-					backgroundColor={ backgroundColor.color }
-					fallbackBackgroundColor={ fallbackBackgroundColor }
-					fallbackTextColor={ fallbackTextColor }
-					fontSize={ fontSize.size }
-					setBackgroundColor={ setBackgroundColor }
-					setTextColor={ setTextColor }
-					textColor={ textColor.color }
-				/>
 			</InspectorControls>
-			<RichText
-				identifier="content"
-				tagName="p"
-				className={ classnames( 'wp-block-paragraph', className, {
-					'has-text-color': textColor.color,
-					'has-background': backgroundColor.color,
-					'has-drop-cap': dropCap,
-					[ `has-text-align-${ align }` ]: align,
-					[ backgroundColor.class ]: backgroundColor.class,
-					[ textColor.class ]: textColor.class,
-					[ fontSize.class ]: fontSize.class,
-				} ) }
-				style={ {
-					backgroundColor: backgroundColor.color,
-					color: textColor.color,
-					fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
-					direction,
-				} }
-				value={ content }
-				onChange={ ( newContent ) => setAttributes( { content: newContent } ) }
-				onSplit={ ( value ) => {
-					if ( ! value ) {
-						return createBlock( name );
-					}
+			{ InspectorControlsColorPanel }
+			<BackgroundColor>
+				<TextColor>
+					<ColorDetector querySelector='[contenteditable="true"]' />
```

Comment:
Is it possible to have these three components just combined into one component `<ApplyColors>` returned by the hook?

When do we need the flexibility?

Reply:
Whenever the targets are not direct siblings, so probably often.

Reply:
I'm not so sure about that, in our current usage, it's always the same target. And if it's a different target, I'd probably expect useColors to be called multiple times.

Reply:
That would create multiple panels which you would then probably have to compose. I think this is unnecessarily limiting the API to drop a few lines of code.

@jorgefilipecosta How often do you think we'll have different color application targets?

Reply:
Are there any docs that explain how all those components created by the `useColors` hooks should be used?

I agree with @youknowriad that it would be much easier to consume it if there were only one component that needs to be wired without too much thinking about the tree structure and placement. Ideally, it would be only
```js
{ InspectorControlsColorPanel }
```
when you assume that everything happens inside `__experimentalUseColors(` which already takes 3 params with different configs.

Reply:
Anyway, if it works, we can always refine it later. It isn't different from what we have in the Heading block. We can refactor all occurrences in core blocks and discuss afterward.

Reply:
Yes, I guess now that we have another usage of useColors and most of the useColors functionality was added we should look into the API format we have now and document it and see if we can increase the developer experience. cc: @epiqueras

Reply:
`InspectorControlsColorPanel` just renders the panel, the components like `BackgroundColor` apply the colors.

I don't think the basic configuration or usage is complex, only the contrast checking configuration is, because it needs to support a lot of use cases.

Documentation will help surface any inconsistencies.